### PR TITLE
fix(popover): fix keyboard arrow navigation

### LINKED ISSUE #23512: bug: Arrow Keys do not seem to work with Popover
# Bug Report

**Ionic version:**
[ ] **4.x**
[ ] **5.x**
[x] **6.x**

**Current behavior:**
When using the keyboard to interact with the Popover the arrow keys do not move up and down the list items. Once the trigger button has focus I can open the menu with the space or enter key and then tab between the items. But up and down arrow does not work.

**Expected behavior:**
The documentation confirms the arrow keys should work for keyboard navigation between items.

image

[https://beta.ionicframework.com/docs/api/popover#keyboard-navigation](https://beta.ionicframework.com/docs/api/popover#keyboard-navigation)

Angular Material opens a select style dropdown with the first item already focused saving a key stroke. It would be good to have such a feature for desktop users.

image

[https://material.angular.io/components/select/overview#select-form](url)

Also the regular Angular Material Menu dropdown can navigate down to nested menus using only arrow keys.
image

[https://material.angular.io/components/menu/examples#menu-icons](https://material.angular.io/components/menu/examples#menu-icons)

**Steps to reproduce:**
Click or set focus to a button that can open a Popover then use the tab key to focus to first item then try to use arrow keys to move up and down between items.

**Related code:**
CodeSandBox Demo using Ionic Custom Elements
[https://codesandbox.io/s/thirsty-brook-b9lhr?file=/src/index.js](https://codesandbox.io/s/thirsty-brook-b9lhr?file=/src/index.js)

**Ionic info:**
Ionic 6.0.0-dev.202106232232.ceabba1

### PR DESCRIPTION
**Pull request by willmartindev**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: resolves #23512 

## What is the new behavior?

- Keyboard navigation now works with arrow keys.
- `Home` and `End` key presses will also navigate to the first and last focusable item, respectively. 

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/popover/utils.ts [hunk 1]
```diff
@@ -314,6 +314,11 @@ export const getPrevItem = (items: HTMLIonItemElement[], currentItem: HTMLElemen
   return items[currentItemIndex - 1];
 }
 
+/** Focus the internal button of the ion-item */
+const focusItem = (item: HTMLIonItemElement) => {
+  (item.shadowRoot?.querySelector('button.item-native') as HTMLElement)?.focus()
```
Comment (liamdebeasi) [inline core/src/components/popover/utils.ts:319]:
```suggestion
  const root = getElementRoot(item);
  const button = root.querySelector('button');
  
  if (button) {
    button.focus();
  }
```

Can we use the `getElementRoot` function here instead? https://github.com/ionic-team/ionic-framework/blob/7315e0157b917d2e3586c64f8082026827812943/core/src/utils/helpers.ts#L97

**Review by liamdebeasi — CHANGES_REQUESTED** [review]
Small code change, but otherwise this looks great. Nice job adding the Home and End keys too.

**Review by liamdebeasi — APPROVED** [review]
Code looks good - nice job. Once the build is passing this is good to merge.

**Comment by willmartindev** [comment]
@liamdebeasi thanks for your help with the test!

**Pull request merged**